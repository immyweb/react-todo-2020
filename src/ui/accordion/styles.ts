import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
