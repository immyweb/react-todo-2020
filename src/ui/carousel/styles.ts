import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
