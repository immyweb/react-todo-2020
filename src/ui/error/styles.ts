import glamorous from 'glamorous';

import { H3 } from '../typography/typography';
import { fonts } from '../../styles/fonts';

interface IGProps {
  backgroundColour?: string;
  marginBottom?: number;
}

export const ErrorWrapper = glamorous.div<IGProps>(
  {
    label: 'error-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Graphic = glamorous.div({
  label: 'graphic',
  position: 'relative',
  margin: '0 auto 10px auto',
});
