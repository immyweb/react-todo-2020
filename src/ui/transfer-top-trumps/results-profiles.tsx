import * as React from 'react';

import ButtonCta from './button-cta';

import {
  ResultsProfileImage,
  ResultsProfileInner,
  ResultsProfileIcon,
  ResultsProfileImageContainer,
} from './styles';
import { colours } from '../../styles/colours';

import { IPlayer } from './types';

export const ResultsProfiles: React.SFC<{ sortedProfiles: IPlayer[] }> = ({ sortedProfiles }) => {
  const profiles = sortedProfiles.map((profile: IPlayer, i: number) => {
    if (profile !== undefined) {
      const { imageMobile, sName, selected } = profile;
      return (
        <ResultsProfileInner key={i}>
          {selected && (
            <ResultsProfileIcon>
              <ButtonCta
                bkgndTop={colours.$lightGreen}
                bkgndBottom={colours.$darkGreen}
                iconName={'tick-solid'}
                size={20}
                borderWidth={1}
                iconSize={8}
                disabled={true}
              />
            </ResultsProfileIcon>
          )}
          {!selected && (
            <ResultsProfileIcon>
              <ButtonCta
                bkgndTop={colours.$lightRed}
                bkgndBottom={colours.$darkRed}
                iconName={'cross-solid'}
                size={20}
                borderWidth={1}
                iconSize={7}
                disabled={true}
              />
            </ResultsProfileIcon>
          )}
          <ResultsProfileImage>
            <img src={imageMobile} alt={sName} />
          </ResultsProfileImage>
        </ResultsProfileInner>
      );
    }
    return <div key={i} />;
  });
  return <ResultsProfileImageContainer>{profiles}</ResultsProfileImageContainer>;
};
