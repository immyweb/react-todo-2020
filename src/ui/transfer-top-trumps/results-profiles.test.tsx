import * as React from 'react';
import { shallow } from 'enzyme';

import { ResultsProfiles } from './results-profiles';
import ButtonCta from './button-cta';
import { ResultsProfileImage } from './styles';

const players = [
  {
    id: 'willian',
    fname: '',
    sName: 'Willian',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg',
    club: 'Chelsea',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png',
    age: 29,
    fee: '£60m',
    position: 'CAM',
    voted: true,
    selected: false,
    active: false,
    idx: 0,
    voteCount: 20,
  },
  {
    id: 'a_sandro',
    fname: 'Alex',
    sName: 'Sandro',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg',
    club: 'Juventus',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png',
    age: 27,
    fee: '£60m',
    position: 'LB',
    voted: true,
    selected: true,
    active: false,
    idx: 1,
    voteCount: 50,
  },
  {
    id: 's_m_savic',
    fname: 'Sergej',
    sName: 'Milinković-Savić',
    imageMobile:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg',
    imageDesktop:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg',
    club: 'SS Lazio',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png',
    age: 23,
    fee: '£70m',
    position: 'CM',
    voted: false,
    selected: false,
    active: true,
    idx: 2,
    voteCount: 60,
  },
];

function shallowRender(props?: any) {
  return shallow(<ResultsProfiles sortedProfiles={players} {...props} />);
}

describe('<ResultsProfiles />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct number of profile images', () => {
    const wrapper = shallowRender();
    const imgs = wrapper.find(ResultsProfileImage);
    expect(imgs.length).toEqual(3);
    const wrapper2 = shallowRender({ sortedProfiles: [] });
    const imgs2 = wrapper2.find(ResultsProfileImage);
    expect(imgs2.length).toEqual(0);
  });

  it('renders yes badge if player has been selected', () => {
    const player = [
      {
        id: 'a_sandro',
        fname: 'Alex',
        sName: 'Sandro',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg',
        imageDesktop: '',
        club: '',
        clubBadge: '',
        age: 27,
        fee: '£60m',
        position: 'LB',
        voted: true,
        selected: true,
        active: false,
        idx: 1,
        voteCount: 50,
      },
    ];
    const wrapper = shallowRender({ sortedProfiles: player });
    const icon = wrapper.find(ButtonCta);
    expect(icon.props().iconName).toEqual('tick-solid');
  });

  it('renders no badge if player has not been selected', () => {
    const player = [
      {
        id: 'a_sandro',
        fname: 'Alex',
        sName: 'Sandro',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg',
        imageDesktop: '',
        club: '',
        clubBadge: '',
        age: 27,
        fee: '£60m',
        position: 'LB',
        voted: true,
        selected: false,
        active: false,
        idx: 1,
        voteCount: 50,
      },
    ];
    const wrapper = shallowRender({ sortedProfiles: player });
    const icon = wrapper.find(ButtonCta);
    expect(icon.props().iconName).toEqual('cross-solid');
  });
});
