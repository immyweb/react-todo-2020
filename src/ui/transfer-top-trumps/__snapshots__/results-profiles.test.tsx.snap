// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsProfiles /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    sortedProfiles={
      Array [
        Object {
          "active": false,
          "age": 29,
          "club": "Chelsea",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png",
          "fee": "£60m",
          "fname": "",
          "id": "willian",
          "idx": 0,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
          "position": "CAM",
          "sName": "Willian",
          "selected": false,
          "voteCount": 20,
          "voted": true,
        },
        Object {
          "active": false,
          "age": 27,
          "club": "Juventus",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png",
          "fee": "£60m",
          "fname": "Alex",
          "id": "a_sandro",
          "idx": 1,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
          "position": "LB",
          "sName": "Sandro",
          "selected": true,
          "voteCount": 50,
          "voted": true,
        },
        Object {
          "active": true,
          "age": 23,
          "club": "SS Lazio",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png",
          "fee": "£70m",
          "fname": "Sergej",
          "id": "s_m_savic",
          "idx": 2,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
          "position": "CM",
          "sName": "Milinković-Savić",
          "selected": false,
          "voteCount": 60,
          "voted": false,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)>
          <glamorous(div)>
            <ButtonCta
              bkgndBottom="#9f041b"
              bkgndTop="#ed283a"
              borderWidth={1}
              disabled={true}
              iconName="cross-solid"
              iconSize={7}
              size={20}
            />
          </glamorous(div)>
          <glamorous(div)>
            <img
              alt="Willian"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
            />
          </glamorous(div)>
        </glamorous(div)>,
        <glamorous(div)>
          <glamorous(div)>
            <ButtonCta
              bkgndBottom="#6cb71b"
              bkgndTop="#a5dc39"
              borderWidth={1}
              disabled={true}
              iconName="tick-solid"
              iconSize={8}
              size={20}
            />
          </glamorous(div)>
          <glamorous(div)>
            <img
              alt="Sandro"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
            />
          </glamorous(div)>
        </glamorous(div)>,
        <glamorous(div)>
          <glamorous(div)>
            <ButtonCta
              bkgndBottom="#9f041b"
              bkgndTop="#ed283a"
              borderWidth={1}
              disabled={true}
              iconName="cross-solid"
              iconSize={7}
              size={20}
            />
          </glamorous(div)>
          <glamorous(div)>
            <img
              alt="Milinković-Savić"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
            />
          </glamorous(div)>
        </glamorous(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            false,
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />
            </glamorous(div)>,
            <glamorous(div)>
              <img
                alt="Willian"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
              />
            </glamorous(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bkgndBottom": "#9f041b",
                "bkgndTop": "#ed283a",
                "borderWidth": 1,
                "disabled": true,
                "iconName": "cross-solid",
                "iconSize": 7,
                "size": 20,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Willian"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Willian",
                "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#6cb71b"
                bkgndTop="#a5dc39"
                borderWidth={1}
                disabled={true}
                iconName="tick-solid"
                iconSize={8}
                size={20}
              />
            </glamorous(div)>,
            false,
            <glamorous(div)>
              <img
                alt="Sandro"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
              />
            </glamorous(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ButtonCta
                bkgndBottom="#6cb71b"
                bkgndTop="#a5dc39"
                borderWidth={1}
                disabled={true}
                iconName="tick-solid"
                iconSize={8}
                size={20}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bkgndBottom": "#6cb71b",
                "bkgndTop": "#a5dc39",
                "borderWidth": 1,
                "disabled": true,
                "iconName": "tick-solid",
                "iconSize": 8,
                "size": 20,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Sandro"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Sandro",
                "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            false,
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />
            </glamorous(div)>,
            <glamorous(div)>
              <img
                alt="Milinković-Savić"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
              />
            </glamorous(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bkgndBottom": "#9f041b",
                "bkgndTop": "#ed283a",
                "borderWidth": 1,
                "disabled": true,
                "iconName": "cross-solid",
                "iconSize": 7,
                "size": 20,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Milinković-Savić"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Milinković-Savić",
                "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)>
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />
            </glamorous(div)>
            <glamorous(div)>
              <img
                alt="Willian"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
              />
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(div)>
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#6cb71b"
                bkgndTop="#a5dc39"
                borderWidth={1}
                disabled={true}
                iconName="tick-solid"
                iconSize={8}
                size={20}
              />
            </glamorous(div)>
            <glamorous(div)>
              <img
                alt="Sandro"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
              />
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(div)>
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                borderWidth={1}
                disabled={true}
                iconName="cross-solid"
                iconSize={7}
                size={20}
              />
            </glamorous(div)>
            <glamorous(div)>
              <img
                alt="Milinković-Savić"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
              />
            </glamorous(div)>
          </glamorous(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              false,
              <glamorous(div)>
                <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  borderWidth={1}
                  disabled={true}
                  iconName="cross-solid"
                  iconSize={7}
                  size={20}
                />
              </glamorous(div)>,
              <glamorous(div)>
                <img
                  alt="Willian"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
                />
              </glamorous(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  borderWidth={1}
                  disabled={true}
                  iconName="cross-solid"
                  iconSize={7}
                  size={20}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bkgndBottom": "#9f041b",
                  "bkgndTop": "#ed283a",
                  "borderWidth": 1,
                  "disabled": true,
                  "iconName": "cross-solid",
                  "iconSize": 7,
                  "size": 20,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Willian"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Willian",
                  "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)>
                <ButtonCta
                  bkgndBottom="#6cb71b"
                  bkgndTop="#a5dc39"
                  borderWidth={1}
                  disabled={true}
                  iconName="tick-solid"
                  iconSize={8}
                  size={20}
                />
              </glamorous(div)>,
              false,
              <glamorous(div)>
                <img
                  alt="Sandro"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
                />
              </glamorous(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ButtonCta
                  bkgndBottom="#6cb71b"
                  bkgndTop="#a5dc39"
                  borderWidth={1}
                  disabled={true}
                  iconName="tick-solid"
                  iconSize={8}
                  size={20}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bkgndBottom": "#6cb71b",
                  "bkgndTop": "#a5dc39",
                  "borderWidth": 1,
                  "disabled": true,
                  "iconName": "tick-solid",
                  "iconSize": 8,
                  "size": 20,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Sandro"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Sandro",
                  "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              false,
              <glamorous(div)>
                <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  borderWidth={1}
                  disabled={true}
                  iconName="cross-solid"
                  iconSize={7}
                  size={20}
                />
              </glamorous(div)>,
              <glamorous(div)>
                <img
                  alt="Milinković-Savić"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
                />
              </glamorous(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  borderWidth={1}
                  disabled={true}
                  iconName="cross-solid"
                  iconSize={7}
                  size={20}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bkgndBottom": "#9f041b",
                  "bkgndTop": "#ed283a",
                  "borderWidth": 1,
                  "disabled": true,
                  "iconName": "cross-solid",
                  "iconSize": 7,
                  "size": 20,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Milinković-Savić"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Milinković-Savić",
                  "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
